driver/src/test/java/com/code_intelligence/jazzer/junit/FuzzingWithCrashTest.java

```diff
@@ -0,0 +1,144 @@
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Comparator;
+import java.util.stream.Stream;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.platform.testkit.engine.EngineExecutionResults;
+import org.junit.platform.testkit.engine.EngineTestKit;
+import org.junit.platform.testkit.engine.EventType;
+import org.opentest4j.AssertionFailedError;
+
+public class FuzzingWithCrashTest {
+  private static final String CRASHING_SEED_NAME = "crashing_seed";
+  // Crashes ByteFuzzTest since 'b' % 2 == 0.
+  private static final byte[] CRASHING_SEED_CONTENT = new byte[] {'b', 'a', 'c'};
+  private static final String CRASHING_SEED_DIGEST = "5e4dec23c9afa48bd5bee3daa2a0ab66e147012b";
+
+  Path seedCorpus;
+
+  @Before
+  public void setup() throws IOException {
+    // Create a fake test resource directory structure with a seed corpus directory in the current
+    // working directory to verify that Jazzer uses it and emits a crash file into it.
+    seedCorpus = Paths.get("src", "test", "resources", "com", "example", "CustomSeedCorpus");
+    Files.createDirectories(seedCorpus);
+    Files.write(seedCorpus.resolve(CRASHING_SEED_NAME), CRASHING_SEED_CONTENT);
+  }
+
+  private static EngineExecutionResults executeTests() {
+    return EngineTestKit.engine("com.code_intelligence.jazzer")
+        .selectors(selectClass("com.example.ValidFuzzTests"))
+        .configurationParameter(
+            "jazzer.instrument", "com.other.package.**,com.example.**,com.yet.another.package.*")
+        .execute();
+  }
+
+  @Test
+  public void fuzzingEnabled() throws IOException {
+    assumeFalse(System.getenv("JAZZER_FUZZ").isEmpty());
+
+    EngineExecutionResults results = executeTests();
+
+    results.testEvents().debug().assertEventsMatchExactly(
+        event(type(EventType.STARTED),
+            test("com.example.ValidFuzzTests", "byteFuzz(byte[]) (Fuzzing)")),
+        event(type(EventType.FINISHED),
+            test("com.example.ValidFuzzTests", "byteFuzz(byte[]) (Fuzzing)"),
+            finishedWithFailure(instanceOf(AssertionFailedError.class))),
+        event(type(EventType.SKIPPED),
+            test("com.example.ValidFuzzTests", "noCrashFuzz(byte[]) (Fuzzing)")),
+        event(type(EventType.SKIPPED),
+            test("com.example.ValidFuzzTests", "dataFuzz(FuzzedDataProvider) (Fuzzing)")));
+    results.containerEvents().debug().assertEventsMatchExactly(
+        event(type(EventType.STARTED), container("com.code_intelligence.jazzer")),
+        event(type(EventType.FINISHED), container("com.code_intelligence.jazzer")));
+
+    // The checks below currently fail on Windows, probably because Bazel doesn't use sandboxing
+    // there and we write into the runfiles tree.
+    assumeFalse(System.getProperty("os.name").startsWith("Windows"));
```

Comment:
Up until now we disabled system specific tests in bazel. Should we do this here as well?

Reply:
We aren't disabling the entire test, we are only disabling some additional checks. I prefer disabling as little as possible since our aim should be to reenable this sooner rather than later. ;-)

Reply:
Right, I see.